src/index.ts

```diff
@@ -0,0 +1,508 @@
+        fired = true;
+        clearInterval(interval);
+        callback();
+      }
+    }, 1000 / this.fps);
+    requestAnimationFrame(() => {
+      if (!fired) {
+        fired = true;
+        clearInterval(interval);
+        callback();
+      }
+    });
+  }
+
+  start() {
+
+    // Hidden canvas element for merging
+    const canvas = this._canvas = document.createElement('canvas');
+    this._canvas.setAttribute('width', this.width.toString());
+    this._canvas.setAttribute('height', this.height.toString());
+    this._canvas.setAttribute('style', 'position:fixed; left: 110%; pointer-events: none'); // Push off screen
+    this._ctx = this._canvas.getContext('2d');
+
+    this.started = true;
+    this._requestAnimationFrame(this._draw.bind(this));
+
+    // Add video
+    this.result = this._canvas?.captureStream(this.fps) || null;
+
+    // Remove "dead" audio track
+    const deadTrack = this.result?.getAudioTracks()[0];
+    if (deadTrack) {this.result?.removeTrack(deadTrack);}
+
+    // Add audio
+    const audioTracks = this._audioDestination?.stream.getAudioTracks();
+    if (audioTracks && audioTracks.length) {
+      this.result?.addTrack(audioTracks[0]);
+    }
+  }
+
+  _updateAudioDelay(delayInMs: number) {
+    if (this._videoSyncDelayNode && this._audioCtx) {
+      this._videoSyncDelayNode.delayTime.setValueAtTime(delayInMs / 1000, this._audioCtx.currentTime);
+    }
+  }
+
+  _draw() {
+    if (!this.started) {return;}
+
+    this._frameCount++;
+
+    // update video processing delay every 60 frames
+    let t0  = 0;
+    if (this._frameCount % 60 === 0) {
+      t0 = performance.now();
+    }
+
+    let awaiting = this._streams.length;
+    const done = () => {
+      awaiting--;
+      if (awaiting <= 0) {
+        if (this._frameCount % 60 === 0) {
+          const t1 = performance.now();
+          this._updateAudioDelay(t1 - t0);
+        }
+        this._requestAnimationFrame(this._draw.bind(this));
+      }
+    };
+
+    if (this.clearRect) {
+      this._ctx?.clearRect(0, 0, this.width, this.height);
+    }
+    this._streams.forEach((stream) => {
+      if (stream.draw) { // custom frame transform
+        stream.draw(this._ctx, stream.element, done);
+      } else if (!stream.isData && stream.hasVideo) {
+        this._drawVideo(stream.element, stream);
+        done();
+      } else {
+        done();
+      }
+    });
+
+    if (this._streams.length === 0) {
+      done();
+    }
+  }
+
+  _drawVideo(element: HTMLVideoElement, stream: any) {
+
+
+    // default draw function
+    const canvasSize = { height: this.height, width: this.width};
+
+    const position = {
+      x: stream.x || 0,
+      y: stream.y || 0
+    };
+
+    const size = {
+        height: stream.height || element.videoHeight || canvasSize.height,
+        width: stream.width || element.videoWidth || canvasSize.width
+    };
+
+    // TODO move to sreeam option to enable new behavior
+    const keepRatio = false;
```

Comment:
I like this feature, moving it into an option is a little awkward though.

Maybe - as the default behaviour - we should maintain the aspect ratio if a width or height isn't specified (and fill the canvas if neither are).

Reply:
I think option may be needed to prevent unexpected new behaviour.
Let me know and i will make changes as you want it to be.

Reply:
New behaviour is fine in a major version (I doubt many applications are using a warped aspect ratio anyways).

If you can remove this feature from this PR, I'll open another one to implement fixed aspect ratio. Don't want to block the rest of your fixes. 👍

Reply:
Ok I will revert this part.
Thank you.